### ISSUE TITLE
SystemJS instead of AMD? 

### ISSUE THREAD

**Issue by vigneshshanmugam** [zalando/tailor#118]
We should start discussing about this soon since SystemJS supports native modules and supports lots of useful feature like custom hooks/resolvers.

**Comment by dazld** [comment]
this would most likely help with getting better error messages out of the bundles too

**Comment by addi90** [comment]
But wasn't the case that native modules are still not supported by browsers? 

Does this ticket mean we handle the module loading for now using SystemJS and once the modules are supported natively by all browsers (I think only Safari supports currently in unstable), utilising browsers for this?

**Comment by vigneshshanmugam** [comment]
Yes native modules are supported only on safari tech preview and MS Edge. The idea here is to load modules using SystemJS instead of AMD and use native modules if its present.

**Comment by addi90** [comment]
Cool. Makes sense

**Comment by dazld** [comment]
a quick look through some issues and the API suggests that we can get the original error, which is going to help a lot with error tracking and hooking into reporting.
